### PR TITLE
WFCORE-6331 Remove deprecated ProductConfig constructor

### PR DESCRIPTION
**Pull request by parsharma**
Issue: https://issues.redhat.com/browse/WFCORE-6331

### CODE REVIEW

**Comment by kabir** [comment]
@parsharma Since the PatchInfoTestCase failures happen across a lot of the WFCOR test runs, it looks like perhaps the test needs adjusting for this change. Apart from that it looks good (i.e. the full integration tests work)

**Review by yersan — CHANGES_REQUESTED** [review]
@parsharma errors are legit, there is a reflection that invokes the constructor you have removed here:

https://github.com/wildfly/wildfly-core/blob/main/testsuite/patching/src/test/java/org/jboss/as/test/patching/ProductInfo.java#L36

As the Java doc states, you should change the reflection call to use `fromFilesystemSlot` instead

File: testsuite/patching/src/test/java/org/jboss/as/test/patching/ProductInfo.java [hunk 1]
```diff
@@ -1,16 +1,15 @@
...
 import org.jboss.modules.LocalModuleLoader;
 import org.jboss.modules.Module;
 import org.jboss.modules.ModuleIdentifier;
 import org.jboss.modules.ModuleLoader;
 
+import java.io.File;
+import java.lang.reflect.Method;
+import java.util.Collections;
+import java.util.Map;
+
```
[the remark this answers is not in the document]
Comment (parsharma) [reply testsuite/patching/src/test/java/org/jboss/as/test/patching/ProductInfo.java:12]: Thanks @bstansberry, I have updated the imports.

**Comment by parsharma** [comment]
Thanks, @yersan & @kabir for the review comments.

**Review by yersan — APPROVED** [review]

**Comment by yersan** [comment]
Thanks @parsharma and @bstansberry / @kabir

**Pull request merged**